File: backend/src/main/java/com/paloit/meeting_room_booking/dbmodel/Room.java
package com.paloit.meeting_room_booking.dbmodel;

import jakarta.persistence.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "room")
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private Integer capacity;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

//    @OneToMany(mappedBy = "room")
//    private List<Booking> bookings;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getCapacity() {
        return capacity;
    }

    public void setCapacity(Integer capacity) {
        this.capacity = capacity;
    }

    public LocalDateTime getCreatedAt() {return createdAt;}

    public void setCreatedAt(LocalDateTime createdAt) {this.createdAt = createdAt;}

    public  LocalDateTime getUpdatedAt() { return  updatedAt;}

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

//    public List<Booking> getBookings() {
//        return bookings;
//    }
//
//    public void setBookings(List<Booking> bookings) {
//        this.bookings = bookings;
//    }
}

File: mobile/lib/model/room.dart
class Room {
  final int? id;
  final String? name;
  final int? capacity;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Room({
    this.id,
    this.name,
    this.capacity,
    this.createdAt,
    this.updatedAt,
  });

  factory Room.fromJson(Map<String, dynamic>? data) {
    return Room(
      id: data?['id'],
      name: data?['name'],
      capacity: data?['capacity'],
      createdAt: DateTime.parse(data?['createdAt']),
      updatedAt: DateTime.parse(data?['updatedAt']),
    );
  }
}
